scheduler/scheduler-api/src/main/java/org/ow2/proactive/scheduler/common/job/factories/model/DefaultModelJobValidatorFactory.java

```diff
@@ -0,0 +1,391 @@
+ * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307
+ * USA
+ *
+ * If needed, contact us to obtain a release under GPL Version 2 or 3
+ * or a different license than the AGPL.
+ *
+ *  Initial developer(s):               The ProActive Team
+ *                        http://proactive.inria.fr/team_members.htm
+ *  Contributor(s):
+ *
+ *  * $$ACTIVEEON_INITIAL_DEV$$
+ */
+package org.ow2.proactive.scheduler.common.job.factories.model;
+
+import com.google.common.base.Strings;
+import org.apache.commons.io.IOUtils;
+import org.apache.log4j.Logger;
+import org.ow2.proactive.scheduler.common.exception.JobCreationException;
+import org.ow2.proactive.scheduler.common.job.TaskFlowJob;
+import org.ow2.proactive.scheduler.common.job.factories.JobValidatorFactory;
+import org.ow2.proactive.scheduler.common.task.Task;
+import org.ow2.proactive.scheduler.common.task.TaskVariable;
+
+import java.io.File;
+import java.io.IOException;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.net.MalformedURLException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.nio.charset.Charset;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.HashSet;
+import java.util.List;
+import java.util.regex.Pattern;
+
+
+/**
+ * TaskFlowJob validator which validates variables against model definition
+ */
+public class DefaultModelJobValidatorFactory implements JobValidatorFactory {
+
+    protected static final Logger logger = Logger.getLogger(DefaultModelJobValidatorFactory.class);
+
+    public static final String INTEGER_MODEL = "INTEGER_MODEL";
+
+    public static final String LONG_MODEL = "LONG";
+
+    public static final String DOUBLE_MODEL = "DOUBLE";
+
+    public static final String BOOLEAN_MODEL = "BOOLEAN";
+
+    public static final String FLOAT_MODEL = "FLOAT";
+
+    public static final String SHORT_MODEL = "SHORT";
+
+    public static final String URL_MODEL = "URL";
+
+    public static final String URI_MODEL = "URI";
+
+    public static final String DATETIME_MODEL = "DATETIME";
+
+    public static final String LIST_MODEL = "LIST";
+
+    public static final String REGEXP_MODEL = "REGEXP";
+
+    public static final String MODEL_FROM_URL_MODEL = "MODEL_FROM_URL";
+
+    public DefaultModelJobValidatorFactory() {
+        // empty
+    }
+
+    @Override
+    public void validateJob(File jobFile) throws JobCreationException {
+        // validate any job
+    }
+
+    @Override
+    public void validateJob(TaskFlowJob job) throws JobCreationException {
+        for (Task task : job.getTasks()) {
+            for (TaskVariable variable : task.getVariables().values()) {
+                checkVariableFormat(task, variable);
+            }
+        }
+    }
+
+    public void checkVariableFormat(Task task, TaskVariable variable) throws JobCreationException {
```

Comment:
Please add a comment to describe what are the expected requirements for the variable model value.